Gitweb:     http://git.kernel.org/linus/bab636b921017f0db6e0c2979438f50b898a9808
Commit:     bab636b921017f0db6e0c2979438f50b898a9808
Parent:     66d0ae4d6ffa45b8e6d8bdbf85f8f1b285c8152d
Author:     Rafael J. Wysocki <rjw@sisk.pl>
AuthorDate: Thu Dec 3 20:21:21 2009 +0100
Committer:  Rafael J. Wysocki <rjw@sisk.pl>
CommitDate: Sun Dec 6 16:17:56 2009 +0100

    PM / Runtime: Fix lockdep warning in __pm_runtime_set_status()
    
    Lockdep complains about taking the parent lock in
    __pm_runtime_set_status(), so mark it as nested.
    
    Signed-off-by: Rafael J. Wysocki <rjw@sisk.pl>
    Reported-by: Alan Stern <stern@rowland.harvard.edu>
    Cc: stable@kernel.org
---
 drivers/base/power/runtime.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

--- a/drivers/base/power/runtime.c
+++ b/drivers/base/power/runtime.c
@@ -777,7 +777,7 @@ int __pm_runtime_set_status(struct devic
 	}
 
 	if (parent) {
-		spin_lock(&parent->power.lock);
+		spin_lock_nested(&parent->power.lock, SINGLE_DEPTH_NESTING);
 
 		/*
 		 * It is invalid to put an active child under a parent that is
